### PR TITLE
add test for own properties on ObjectWrap

### LINKED ISSUE #644: ObjectWrap: enumerable properties seem to be broken
in 'test/objectwrap.cc' the property 'testGetter' is defined as enumerable:
```C++
InstanceAccessor("testGetter", &Test::Getter, nullptr, napi_enumerable),
```

but testing if this property is enumerable:

```bash
gms@orion:~/work/RESEARCH/node-addon-api (master *)$ git diff
diff --git a/test/objectwrap.js b/test/objectwrap.js
index de16a60..e5d3efc 100644
--- a/test/objectwrap.js
+++ b/test/objectwrap.js
@@ -11,6 +11,10 @@ const test = (binding) => {
   };
 
   const testAccessor = (obj, clazz) => {
+    // enumerable
+    {
+      assert.strictEqual(obj.propertyIsEnumerable('testGetter'), true);
+    }
     // read-only, write-only
     {
       obj.testSetter = 'instance getter';

```

fails with:
```bash
AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:

false !== true
```

### PR DESCRIPTION
**Pull request by gms1**
fixes #644

### CODE REVIEW

File: test/objectwrap.cc [hunk 1]
```diff
@@ -28,7 +28,22 @@ class Test : public Napi::ObjectWrap<Test> {
     if(info.Length() > 0) {
       finalizeCb_ = Napi::Persistent(info[0].As<Napi::Function>());
     }
+    // Create an own instance property.
+    info.This().As<Napi::Object>().DefineProperty(
+        Napi::PropertyDescriptor::Accessor(info.Env(),
+                                           info.This().As<Napi::Object>(),
+                                           "ownProperty",
+                                           OwnPropertyGetter,
```
Comment (gms1) [inline test/objectwrap.cc]: word considering:
```diff
-                                           OwnPropertyGetter,
-                                           napi_enumerable, this));
+                                           std::bind(&Test::Getter, this, std::placeholders::_1),
+                                           napi_enumerable));
```

File: test/objectwrap.js [hunk 2]
```diff
@@ -84,10 +96,12 @@ const test = (binding) => {
   };
 
   const testEnumerables = (obj, clazz) => {
-    // Object.keys: only object
-    assert.deepEqual(Object.keys(obj), []);
+    // Object.keys: only object without prototype
+    assert(Object.keys(obj).length === 2);
+    assert(Object.keys(obj).indexOf('ownProperty') >= 0);
```
Comment (legendecas) [inline test/objectwrap.js]:
```suggestion
    assert(Object.keys(obj).includes('ownProperty'));
```

**Comment by mhdawson** [comment]
@gms1 if you pull in @legendecas suggested change we can move forward in landing this.

**Comment by gabrielschulhof** [comment]
CI:

| Version | Job | Status |
|---:|---|:---:|
| v14.x | https://ci.nodejs.org/job/node-test-node-addon-api-new/1592/ | ✓ |
| v12.x | https://ci.nodejs.org/job/node-test-node-addon-api-new/1593/ | ✓ |
| v10.x | https://ci.nodejs.org/job/node-test-node-addon-api-new/1594/ | ✓ |

**Comment by gabrielschulhof** [comment]
Landed in e8935bd8d97ba61cd4c005fab571d9fb39ab9249.

**Pull request closed without merging**